Add section explaining table metadata (#774)
astronomer/astro-sdk: python-sdk/docs/concepts.rst

           :start-after: [START temp_table_example]
           :end-before: [END temp_table_example]

Metadata

> ~~~~~~~~~

Reviewer: ```suggestion
~~~~~~~~
```

Metadata is used in tables to pass additional information to access a SQL Table, which will be database specific.

> For example we can have multiple tables in different schema or database, we can pass metadata object while defining table to pass this information.

Reviewer 2: ```suggestion
For example, a user can detail the Snowflake schema and database for a table, whereas a BigQuery user can specify the namespace and dataset. Although these parameters can change name depending on the database, we have normalised the `Metadata` class to name their schema and database.
```

> For example, a user can detail the Snowflake schema and database for a table, whereas a BigQuery user can specify the namespace and dataset. Although these parameters can change name depending on the database, we have normalised the `Metadata` class to name their schema and database.

Reviewer 3: ```suggestion
For example, a user can detail the Snowflake schema and database for a table, whereas a BigQuery user can specify the namespace and dataset. Although these parameters can change name depending on the database, we have normalised the ``Metadata`` class to name their schema and database.
```

or just cross reference Metadata object

Reviewer 3: :class:`~astro.sql.table.Metadata`

.. literalinclude:: ../example_dags/example_amazon_s3_snowflake_transform.py
   :language: python

>    :start-after: [START metadata_example_1]

Reviewer 4: ```suggestion
   :start-after: [START metadata_example_snowflake]
```

>    :end-before: [END metadata_example_1]

Reviewer 4: ```suggestion
   :end-before: [END metadata_example_snowflake]
```
